src/ppopt/utils/region_overlap_utils.py

```diff
@@ -0,0 +1,188 @@
+        inner_region = cr_new_bounds(inner_region, lb_new=None, ub_new=intersection)
+        new_regions[-1] = cr_new_bounds(new_regions[-1], lb_new=intersection, ub_new=None)
+    return new_regions, inner_region, outer_region
+
+
+def split_outer_region(new_regions: List[CriticalRegion], cr: CriticalRegion, inner_lb: float, inner_ub: float) -> \
+        Tuple[List[CriticalRegion], CriticalRegion]:
+    new_regions.append(copy.deepcopy(cr))
+    cr = cr_new_bounds(cr, lb_new=None, ub_new=inner_lb)
+    new_regions[-1] = cr_new_bounds(new_regions[-1], lb_new=inner_ub, ub_new=None)
+    return new_regions, cr
+
+
+def cr_new_bounds(cr: CriticalRegion, lb_new: Optional[float], ub_new: Optional[float]) -> CriticalRegion:
+    lb, ub = get_bounds_1d(cr.E, cr.f)
+
+    lb = lb if lb_new is None else lb_new
+    ub = ub if ub_new is None else ub_new
+
+    cr.E = numpy.array([[1], [-1]])
+    cr.f = numpy.array([[ub], [-lb]])
+
+    return cr
+
+
+def full_overlap(cr1: CriticalRegion, cr2: CriticalRegion) -> bool:
+    # region 2 fully inside region 1
+    lb1, ub1 = get_bounds_1d(cr1.E, cr1.f)
+    lb2, ub2 = get_bounds_1d(cr2.E, cr2.f)
+    return lb1 <= lb2 and ub1 >= ub2
+
+
+def partial_overlap(cr1: CriticalRegion, cr2: CriticalRegion) -> bool:
+    # region 1 to the left of region 2
+    lb1, ub1 = get_bounds_1d(cr1.E, cr1.f)
+    lb2, ub2 = get_bounds_1d(cr2.E, cr2.f)
+    return lb1 < lb2 and ub1 > lb2 and ub2 > ub1
+
+
+def equal_linear_objective(program: MPMILP_Program, inner_region: CriticalRegion, outer_region: CriticalRegion) -> bool:
+    f_inner_lb, f_inner_ub, f_outer_lb, f_outer_ub = evaluate_objective_at_middle_bounds(program, inner_region,
+                                                                                         outer_region)
+    return f_inner_lb == f_outer_lb and f_inner_ub == f_outer_ub
+
+
+def region_2_dominates(program: MPMILP_Program, cr_1: CriticalRegion, cr_2: CriticalRegion) -> bool:
+    f_1_lower, f_1_upper, f_2_lower, f_2_upper = evaluate_objective_at_middle_bounds(program, cr_1, cr_2)
+    return f_1_lower >= f_2_lower and f_1_upper >= f_2_upper
+
+
+def region_1_dominates(program: MPMILP_Program, cr_1: CriticalRegion, cr_2: CriticalRegion) -> bool:
+    f_1_lower, f_1_upper, f_2_lower, f_2_upper = evaluate_objective_at_middle_bounds(program, cr_1, cr_2)
+    return f_1_lower <= f_2_lower and f_1_upper <= f_2_upper
+
+
+# cr_1 is the outer region or the left region, depending on calling context
+def compute_objective_intersection_point(program: MPMILP_Program, cr_1: CriticalRegion, cr_2: CriticalRegion) -> Tuple[
+    float, bool]:
+    f_1_lower, f_1_upper, f_2_lower, f_2_upper = evaluate_objective_at_middle_bounds(program, cr_1, cr_2)
+    deltaf_1 = f_1_upper - f_1_lower
+    deltaf_outer = f_2_upper - f_2_lower
+    lower, upper = find_middle_bounds(cr_1, cr_2)
+    delta = upper - lower
+    return (f_2_lower - f_1_lower) / (deltaf_1 - deltaf_outer) * delta + lower, f_1_lower < f_2_lower
```

Comment:
Here instead of computing the return value in the return line, we can make a variable with a name so it is clear what it is that we are returning